test/src/unittests/tonal/test_tonicindianartmusic.py

```diff
@@ -33,66 +34,88 @@ def testInvalidParam(self):
         self.assertConfigureFails(TonicIndianArtMusic(), { 'harmonicWeight': 1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'hopSize': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeCompression': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeCompression': 2 })        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeThreshold': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'maxTonicFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'minTonicFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': 0 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': -1 })        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberSaliencePeaks': 0})
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'numberSaliencePeaks': 16})        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'referenceFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'sampleRate':   -1 })
 
-    def testZeros(self):
-        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()(zeros(1024)))   
+    def testEmpty(self):
+        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()([]))   
 
-    def testOnes(self):
-        referenceTonic =108.86099243164062     
-        tonic   = TonicIndianArtMusic()(ones(1024))
-        self.assertAlmostEqual(referenceTonic, tonic, 8)
+    def testSilence(self):
+        silence = np.zeros(int(np.abs(np.random.randn()) * 30. * 44100))
+        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()(silence))
 
-    def testNegativeInput(self):
-        tonic = TonicIndianArtMusic()([-1]*1024)
-        referenceTonic =108.86099243164062     
-        tonic   = TonicIndianArtMusic()([-1]*1024)
-        self.assertAlmostEqual(referenceTonic, tonic, 8)
+    def testOnes(self):
+        # FIXME: The tonic value 117.36 is not returned everytime. Sometimes 55 is returned.
+        # Also, different values can be obtained when tweaking minTonic, maxTonic and reference frequencies.
+        referenceTonic = 117.36 
+        tonic   = TonicIndianArtMusic()(ones(4096))
+        self.assertAlmostEqual(tonic, referenceTonic, 5)
 
     def testRegression(self):
         audio = MonoLoader(filename = join(testdata.audio_dir, 'recorded/vignesh.wav'),
                             sampleRate = 44100)()
         referenceTonic = 102.74                                       
         tonic = TonicIndianArtMusic()(audio)
-        self.assertAlmostEqual(referenceTonic, tonic, 6)
+        self.assertAlmostEqual( tonic, referenceTonic, 6)
```

Comment:
`6` is a very large deviation, how do you decide the value? what are the estimated vs reference values?